java/code/src/com/suse/manager/reactor/messaging/RegisterMinionEventMessageAction.java

```diff
@@ -403,6 +403,8 @@ else if (!minion.getOrg().equals(org)) {
                 return;
             }
 
+            MinionList minionTarget = new MinionList(minionId);
+            systemQuery.syncAll(minionTarget);
```

Comment:
Isn't this doing a synchronous Salt call reaching out to that minion? Someone who is an expert on the registration code and it's scalability should probably approve this at least, I remember that previously we tried to avoid doing synchronous calls to Salt during the registration as there is a risk of blocking a thread in case the minion is not responding. If there is any option to run this asynchronously it would definitely be preferable from my perspective.

Reply:
Hmmm, yeah, this would be a `callSync` execution in the end. I'll think how we could make this asynchronous :+1:

Reply:
As discussed, I've removed this call during minion registration. After latest SLS refactoring, the custom `mgrcompat` state module should be automatically deployed while applying any SLS that requires it, particularly, the "Hardware Refresh" action triggered at the beginning of registration will also sync the states.